File: video-api/src/main/java/com/fiapql/videoapi/repository/VideoRepository.java
package com.fiapql.videoapi.repository;

import com.fiapql.videoapi.entity.Video;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;
import java.util.UUID;

public interface VideoRepository extends JpaRepository<Video, UUID> {
    List<Video> findByUserIdOrderByCreatedAtDesc(String userId);
}

File: video-api/src/main/kotlin/com/fiapql/videoapi/repository/VideoRepository.kt
package com.fiapql.videoapi.repository

import com.fiapql.videoapi.entity.Video
import org.springframework.data.jpa.repository.JpaRepository
import java.util.UUID

interface VideoRepository : JpaRepository<Video, UUID> {
    fun findByUserIdOrderByCreatedAtDesc(userId: String): List<Video>
}
